### PR TITLE
config: addition of INDEXER_BEFORE_INDEX_HOOKS

### PR DESCRIPTION
**Pull request by zazasa**
In the init_app, connects functions provided in the INDEXER_BEFORE_INDEX_HOOKS config variable to the before_record_index signal.

--
addresses #46

### CODE REVIEW

**Comment by jirikuncar** [comment]
@zazasa please amend the commit message to follow http://invenio.readthedocs.io/en/latest/technology/git.html#r2-remarks-on-commit-log-messages

File: invenio_indexer/ext.py [hunk 1]
```diff
@@ -52,6 +53,13 @@ def init_app(self, app):
         self.init_config(app)
         app.extensions['invenio-indexer'] = self
 
+        hooks = app.config.get('INDEXER_BEFORE_INDEX_HOOKS', [])
+        for hook in hooks:
+            if callable(hook):
+                before_record_index.connect_via(app)(hook)
+            elif isinstance(hook, str):
```
Comment (jirikuncar) [inline invenio_indexer/ext.py]: @zazasa please make sure that it's both PY2/3 friendly. See https://github.com/inveniosoftware/invenio-records-rest/blob/master/invenio_records_rest/utils.py#L76-L87

File: invenio_indexer/ext.py [hunk 2]
```diff
@@ -52,6 +54,15 @@ def init_app(self, app):
         self.init_config(app)
         app.extensions['invenio-indexer'] = self
 
+        hooks = app.config.get('INDEXER_BEFORE_INDEX_HOOKS', [])
+        for hook in hooks:
+            if callable(hook):
+                before_record_index.connect_via(app)(hook)
+            elif isinstance(hook, six.string_types):
```
Comment (jirikuncar) [inline invenio_indexer/ext.py]: What about simply:
```python
if isinstance(hook, six.string_types):
    hook= import_string(hook)
before_record_index.connect_via(app)(hook)
```

Comment (jirikuncar) [inline invenio_indexer/ext.py]: What about simply:
```python
if isinstance(hook, six.string_types):
    hook= import_string(hook)
before_record_index.connect_via(app)(hook)
```

**Comment by jirikuncar** [comment]
LGTM

**Comment by jirikuncar** [comment]
LGTM

**Comment by jirikuncar** [comment]
@zazasa I have added tests and documentation. Please have a look at it. Thanks!

**Comment by lnielsen** [comment]
LGTM

**Comment by zazasa** [comment]
LGTM

**Pull request merged**